src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/utils/JobTemplateResolutionUtil.scala

```diff
@@ -15,13 +15,17 @@
 
 package za.co.absa.hyperdrive.trigger.api.rest.utils
 
+import org.springframework.stereotype.Component
 import za.co.absa.hyperdrive.trigger.models._
 import za.co.absa.hyperdrive.trigger.configuration.application.JobDefinitionConfig.{KeysToMerge, MergedValuesSeparator}
+import za.co.absa.hyperdrive.trigger.configuration.application.{ShellExecutorConfig, SparkYarnSinkConfig}
 
+import java.nio.file.Paths
+import javax.inject.Inject
 import scala.util.{Failure, Success, Try}
 
-
-object JobTemplateResolutionUtil {
+@Component
+class JobTemplateResolutionUtil @Inject()(sparkYarnSinkConfig: SparkYarnSinkConfig, shellExecutorConfig: ShellExecutorConfig) {
```

Comment:
I think you should rename the class so it doesn't end in `Util` and also move it to some other package, since utils are supposed to be singleton objects. Otherwise, you can pass the executables folder variables.

Reply:
Ok, I made service of it